smoderp2d/providers/base/data_preparation.py

```diff
@@ -260,6 +293,30 @@ def __init__(self, writter):
             self.fieldnames['channel_q365']
         ]
 
+    def _check_parameter_value(self, rr, rc, name, arr, range_):
+        """ check the parameter margins 
+
+        :param list rr: list of i indices in the computaion domain
+        :param list rc: list of j indices in the computaion domain
+        :param str name: name of the variable
+        :param np.array arr: the array holding the parameter values 
+        :param list range_: range of appropriate parameters
+        """
+
+        for i in rr:
+            for j in rc[i]:
+                val = arr[i][j]
+                if (range_[0] > val) : raise SmallParameterValueError(name, val, range_[0])
+                if (range_[1] < val) : raise LargeParameterValueError(name, val, range_[1])
+
+        Logger.info('{} parameter values checked.'.format(name))
+
+
+    def _set_output_data(self):
```

Comment:
where is this new method called?

Reply:
I guess it is more of a rhetoric question, but the answer is nowhere (at least at this moment).